lib/path.js

```diff
@@ -687,6 +707,19 @@ const win32 = {
     if (typeof path !== 'string' || path.length === 0)
       return path;
 
+    // Check if the path matches any device pattern
+    if (ObjectValues(windowDevicePatterns).some((pattern) => pattern.test(path))) {
```

Comment:
Would it be faster if the object was an array?

Reply:
Yes, you're correct. My local benchmarks confirm that using an array is indeed faster than using an object. Thank you very much for your thorough review. I will take this and apply the changes accordingly